### ISSUE TITLE
Unable to replicate results for Norm Aware Latent Space Exploration

### ISSUE THREAD

**Issue by 1392001sai** [dvirsamuel/SeedSelect#4]
Hi, Thank you for the great work.

I am trying to replicate the latent interpolation results on the ImageNet data with the given code. I am observing that the latents obtained from DDIM inversion are not able to recreate the original images exactly when fed back into the stable diffusion model. 

For example for these real images from the paper -
![image](https://github.com/dvirsamuel/SeedSelect/assets/57600628/29f65326-0270-4d55-be28-f079db73cbf7)

If I denoise the DDIM inversion latents generated with the prompt - "tiger cat". I get these images -
![image](https://github.com/dvirsamuel/SeedSelect/assets/57600628/86d899a3-cc59-4bcb-ada5-78606e4af1a7)

I am able to recreate the reference images if I also optimize the unconditional embeddings using Null Text Inversion -
![image](https://github.com/dvirsamuel/SeedSelect/assets/57600628/f86719eb-cea0-487b-a989-bbab81b52fcc)

But with this I would be unable to interpolate between the latents because I don't have the unconditional embeddings corresponding to the interpolated latents.

Could you kindly tell me where I am going wrong and how you generated this interpolation here -
![image](https://github.com/dvirsamuel/SeedSelect/assets/57600628/b1f37c83-d535-4685-bbfa-f03fe9485323)

**Comment by dvirsamuel** [comment]
Hi, thank you for your interest in our work.

The guidance scale should be set to 1 for both ddim inversion and inference/generation. Then, the null-text optimization would not be required and interpolation would be possible.

**Comment by 1392001sai** [comment]
Thank you for the quick response. That fixed it.

**Comment by 1392001sai** [comment]
Hi,

Had a small doubt in the objective function used to calculate loss in norm_aware_optimization.py

```
# Define the objective function
def objective(p1, p2, path, f, eps):
    # Compute the distances between consecutive x's
    dist_1 = torch.unsqueeze(torch.norm(p1 - path[1, :].reshape(p1.shape)), dim=-1)
    dists = torch.norm(path[1:, :] - path[:-1, :], dim=1)
    dist_n = torch.unsqueeze(torch.norm(p2 - path[-1, :].reshape(p2.shape)), dim=-1)
    dists = torch.cat((dist_1, dists, dist_n), dim=0)
```
Here the path variable already excludes the initial points. 

```
def path_between_two_points(a, b, n):
    d = a.shape[0]
    # Initialize the path points to be regularly placed on the cord between a and b
    x = torch.linspace(0, 1, n, dtype=torch.float).view(n, 1).repeat(1, d).to("cuda")
    x = x * (b.t() - a.t()) + a.t()

    # x = x + torch.rand(x.shape).to("cuda") / 100
    x = nn.Parameter(x[1:-1, :])  # without a and b(stay in place)
    return x
```

So, the first point in the path is being skipped.  This is what I get when I print dists.

```
tensor([9.2297, 4.6148, 4.6148, 4.6148, 4.6148, 4.6148, 4.6148, 4.6148, 4.6148,
        4.6148, 4.6148, 4.6148, 4.6149, 4.6148, 4.6148, 4.6149, 4.6148, 4.6149,
        4.6148], device='cuda:0', grad_fn=<CatBackward0>)
```

So shouldn't the dist calculation start from path[0,:] instead?

```
# Define the objective function
def objective(p1, p2, path, f, eps):
    # Compute the distances between consecutive x's
    dist_1 = torch.unsqueeze(torch.norm(p1 - path[0, :].reshape(p1.shape)), dim=-1)
    dists = torch.norm(path[0:-1, :] - path[1:, :], dim=1)
    dist_n = torch.unsqueeze(torch.norm(p2 - path[-1, :].reshape(p2.shape)), dim=-1)
    dists = torch.cat((dist_1, dists, dist_n), dim=0)
```

**Comment by dvirsamuel** [comment]
Hi,
Sorry for the late response.
I am really thankful for your comment. I have changed it in the code repo.

Thanks again, 
Dvir
